homeassistant/components/mqtt/mixins.py

```diff
@@ -1218,8 +1225,6 @@ async def async_added_to_hass(self) -> None:
         self._prepare_subscribe_topics()
```

Comment:
This never actually yields to the loop so it runs synchronously.

The test only passed because the task was started non-eagerly.